pup-0002.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Docstring Style Conventions
+author: Austin Macdonald
+created: 28-Mar-2017
+status: proposed
+---
+
+## Summary
+
+This PUP proposes the use of Google style docstrings for Pulp 3+.
+
+## Motivation
+
+To improve docstring readability and increase information density.
+
+## Details
+
+Upon acceptance, these changes will be documented in the Pulp style guide.
+
+### Style Choices
+
+Docstrings should follow the conventions documented in the
+[Google style guide](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+except where Pulp specific additions or exceptions apply.
+
+Additionally, docstrings must be parsable by the
+[Napoleon plugin for Sphinx.](http://www.sphinx-doc.org/en/stable/ext/napoleon.html)
+
+### Additions/Exceptions
+
+Pulp conventions that differ from the `Comments` section of the Google style guide will be
+explicitly included in the Pulp style guide.
+
+1. Modules will not contain license information. Our license information is presented correctly
+within the project; it is unnecessary to include it in each file.
+
+2. The type of each `Arg` value should be included after the variable name in parentheses. The type of each `Returns` value should be the first item on the line.
+
+3. Following the type of `Arg` and `Return` values, there will be a colon and a single space
+followed by the description. Additional spaces should not be used to align types and descriptions.
+
+4. `Fields` and `Relations` sections will be used when documenting fields on Django models.
+The `Fields` section will be used for non-related fields on Model classes, the `Relations`
+section will be used for related fields on Model classes.
+
+### Porting from Pulp 2.Y
+
+The 3.0 development guide will be updated to state that all ported code should contain updated
+docstrings. Regular expressions to ease large changes will be included there.
+
+## Drawbacks
+
+It will take time to convert each docstrings into the new style.
+
+## Alternatives
```

Comment:
I agree in the general case, but for these specific alternatives, the pros and cons seem obvious and in the realm of personal taste. Do you have pros and cons that you would like to see added?